### PR TITLE
fix(dynamite): Fix nullable someOf parameters

### PR DESCRIPTION
**Pull request by Leptopoda**
#1256

### CODE REVIEW

File: packages/dynamite/dynamite_end_to_end_test/test/parameters_test.dart [hunk 1]
```diff
@@ -121,4 +121,30 @@ void main() {
...
+      await client.getSomeOf(anyOf: ($bool: true, string: null));
+      final captured = verify(mockHttpClient.openUrl(captureAny, captureAny)).captured;
+      expect(
+        captured,
+        equals([
+          'get',
+          Uri.parse('example.com/some-of?anyOf=true'),
+        ]),
+      );
+    });
```
Comment (provokateurin) [inline packages/dynamite/dynamite_end_to_end_test/test/parameters_test.dart:148]: Please also test that it works with a string.

**Review by provokateurin — CHANGES_REQUESTED** [review]

File: packages/dynamite/dynamite/lib/src/builder/client.dart [hunk 2]
```diff
@@ -253,7 +253,11 @@ Iterable<Method> buildTags(
           ),
           parameter.schema!,
           nullable: dartParameterNullable,
-        ).dartType;
+        );
+
+        if (result is! TypeResultSomeOf) {
+          result = result.dartType;
+        }
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/builder/client.dart:260]: Please add a comment that explains why this is needed.

**Comment by Leptopoda** [comment]
I think that it's probably easier to just call serialize (and not encode) on the typeresult and let the uri template handle the rest #1097.
I'll give it a try.

**Comment by provokateurin** [comment]
That sounds good yeah

**Comment by provokateurin** [comment]
yet another problem related to parameter serialization: pattern checking does not work on string values. I would argue it is useless because it is an enum already, but it is a valid spec and also happens to appear in Talk.

**Comment by Leptopoda** [comment]
Pattern checking requires a bigger makeover.
Can you report an example spec to #1247?
Or open a new issue and mention it from there.

**Comment by provokateurin** [comment]
I commented on the issue, but I feel like the problem is just the parameter serialization. The value needs to be serialized before checking the pattern (would be at least one way to fix it).

**Comment by Leptopoda** [comment]
Still bases on #1292

**Review by provokateurin — APPROVED** [review]

**Pull request merged**